src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/_validators.py

```diff
@@ -38,7 +38,7 @@ def internal_validate_lock_parameters(resource_group_name, resource_provider_nam
             raise CLIError('--resource-type is ignored if --resource-group is not given.')
         if resource_provider_namespace is not None:
             raise CLIError('--namespace is ignored if --resource-group is not given.')
-        if parent_resource_path is not None:
+        if parent_resource_path and parent_resource_path is not None:
```

Comment:
If `parent_resource_path == None` then `if parent_resource_path` would return false anyways. So you could just simplify this to `if parent_resource_path is not None`.